framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -121,23 +134,19 @@ PointSamplerBase::finalize()
   // Save off for speed
   const auto pid = processor_id();
 
-  /*
-   * Figure out which processor is actually going "claim" each point.
-   * If multiple processors found the point and computed values what happens is that
-   * maxloc will give us the smallest PID in max_id
-   */
-  std::vector<unsigned int> max_id(_found_points.size());
-
-  _communicator.maxloc(_found_points, max_id);
+  // Consolidate _found_points across processes to know which points were found
+  auto _global_found_points = _found_points;
+  _comm.sum(_global_found_points);
 
-  for (MooseIndex(max_id) i = 0; i < max_id.size(); ++i)
+  for (MooseIndex(_found_points) i = 0; i < _found_points.size(); ++i)
   {
-    // Only do this check on the proc zero because it's the same on every processor
-    // _found_points should contain all 1's at this point (ie every point was found by a proc)
-    if (pid == 0 && !_found_points[i])
+    // _global_found_points should contain all 1's at this point (ie every point was found by a
+    // proc)
+    if (pid == 0 && !_global_found_points[i])
       mooseError("In ", name(), ", sample point not found: ", _points[i]);
 
-    if (max_id[i] == pid)
+    // only process that found the point has the value
+    if (_found_points[i])
```

Comment:
I believe you don't want to change this - you only want one processor to add a sample per point. With the changes you've made, if a point is on a boundary, you will have it show it up multiple times in your sampling.

Reply:
we still have that. _found_points should be local, in which case only the process owning the elem with the lowest ID globally

but you are right for continuous variables where we dont do that check. I ll change it

Reply:
Yeah: multiple processors can potentially set `_found_points` to true if it's on the face of a continuous variable that is also on a processor boundary.